src/base/random.hh

```diff
@@ -99,18 +99,22 @@ class Random
     random()
     {
         // [0, 1) for real types
-        std::uniform_real_distribution<T> dist;
-        return dist(gen);
+        warn("FP random numbers are not uniformly distributed.");
+        return (T) random<uint64_t>() /
```

Comment:
The call to random<uint64_t> adds an unnecessary modulo operation: (gen() % max<uint64_t>) / max<uint64_t>. Any reason not to use gen() directly? Also, could you add parenthesis to where the T casts are being applied to make reading easier?